cf-custom-resources/lib/nlb-cert-validator-updater.js

```diff
@@ -400,15 +413,166 @@ async function activateOption(option, loadBalancerDNS, loadBalancerHostedZone) {
+                Values: [appName],
+            },
+            {
+                Key: "copilot-environment",
+                Values: [envName],
+            },
+            {
+                Key: "copilot-service",
+                Values: [serviceName],
+            }
+        ],
+        ResourceTypeFilters: ["acm:certificate"]
+    }).promise();
+
+    cachedServiceCertificates = [];
+    let promises = [];
+    for (const {ResourceARN: arn} of ResourceTagMappingList) {
+        let promise = clients.acm().describeCertificate({
+            CertificateArn: arn
+        }).promise().then(( { Certificate } ) => {
+            cachedServiceCertificates.push(Certificate);
+        });
+        promises.push(promise);
+    }
+    await Promise.all(promises);
+    return cachedServiceCertificates;
+}
+
+/**
+ * Retrieve the validation options that are pending deletion. An option is pending deletion if it is only used to
+ * validate a certificate that is pending deletion.
+ * @param {Array<Object>} certsPendingDeletion
+ * @param {Array<Object>} certsInUse
+ * @returns {Promise<Set<Object>>} options that are pending deletion.
+ */
+async function unusedOptionsByService(certsPendingDeletion, certsInUse) {
+    let optionsPendingDeletion = new Map();
+    for (const { DomainValidationOptions: validationOptions } of certsPendingDeletion) {
+        for (const option of validationOptions) {
+            optionsPendingDeletion.set(JSON.stringify(option), option);
+        }
+    }
+    for (const { DomainValidationOptions: validationOptions } of certsInUse) {
+        for (const option of validationOptions) {
+            optionsPendingDeletion.delete(JSON.stringify(option));
+        }
+    }
+    let options = new Set();
+    for (const opt of optionsPendingDeletion.values()) {
+        options.add(opt);
+    }
+    return options;
+}
+
+/**
+ * Validate if the domain name is currently in use by other services.
+ * @param loadBalancerDNS The DNS of the Network Load Balancer used by this service. The domain name in considered in use
+ * by this service, not other services, if it is an alias target pointing to this service's load balancer DNS.
+ * @param domainName 
+ * @param route53Client The Route53 client to use for the domain name. This client can be a app-level client, or an
+ * env-level client, depending on the pattern of the domain name. It is initialized outside of the function because
+ * AWS-SDK mocks cannot mock the API call if the client is initialized in a callback.
+ * @returns {Promise<boolean>} True if it is considered in use; otherwise false.
+ */
+async function inUseByOtherServices(loadBalancerDNS, domainName, route53Client) {
+    let hostedZoneID;
+    try {
+        ({hostedZoneID} = await domainResources(domainName));
+    } catch (err) {
+        if (err instanceof UnrecognizedDomainTypeError) {
+            return true; // This option has unrecognized pattern, we can't check if it is in use, so we assume it is in use.
```

Comment:
Should we log the error here just in case to troubleshoot? I think we log generic exceptions but it might be also information to log this one as it's suprising

Reply:
Makes sense!

I think this could happen very rarely, given that the certificate is Copilot-managed. It wouldn't have passed cli level validation in the first place. 

I have added the log!